### PR TITLE
Add support for zipped credentials

### PR DESCRIPTION
**Pull request by OYTIS**

### CODE REVIEW

File: classes/image_types_ostree.bbclass [hunk 1]
```diff
@@ -114,6 +145,23 @@ IMAGE_CMD_ostree () {
...
+		EXPDATE=`openssl pkcs12 -in ${SOTA_AUTOPROVISION_CREDENTIALS} -password "pass:" -nodes 2>/dev/null | openssl x509 -noout -enddate | cut -f2 -d "="`
+
+		if [ `date +%s` -ge `date -d "${EXPDATE}" +%s` ]; then
+			bberror "Certificate ${SOTA_AUTOPROVISION_CREDENTIALS} has expired on ${EXPDATE}"
+		fi
+
+		mkdir -p var/sota
+		cp ${SOTA_AUTOPROVISION_CREDENTIALS} var/sota/sota_provisioning_credentials.p12
+		if [ -n "${SOTA_AUTOPROVISION_URL_FILE}" ]; then
+			export SOTA_AUTOPROVISION_URL="https://`cat ${SOTA_AUTOPROVISION_URL_FILE}`:9000"
```
Comment (cajun-rat) [inline classes/image_types_ostree.bbclass]: I'm not sure that this is the right port in all cases.  I've send an email around to check.

**Comment by OYTIS** [comment]
Changed after the change in the garage.

**Comment by cajun-rat** [comment]
I think this still has the explicit port 9000 in it?

File: classes/image_types_ostree.bbclass [hunk 2]
```diff
@@ -16,6 +17,36 @@ OSTREE_KERNEL ??= "${KERNEL_IMAGETYPE}"
 
 export SYSTEMD_USED = "${@bb.utils.contains('DISTRO_FEATURES', 'systemd', 'true', '', d)}"
 
+python () {
+    if d.getVar("SOTA_PACKED_CREDENTIALS", True):
+        if d.getVar("SOTA_AUTOPROVISION_CREDENTIALS", True):
+            bb.warn("SOTA_AUTOPROVISION_CREDENTIALS are overriden by those in SOTA_PACKED_CREDENTIALS")
+        if d.getVar("SOTA_AUTOPROVISION_URL", True):
+            bb.warn("SOTA_AUTOPROVISION_URL is overriden by one in SOTA_PACKED_CREDENTIALS")
```
Comment (tkfu) [inline classes/image_types_ostree.bbclass]: Typo in warning message. Should be "...is **overridden** by **the** one in..."

**Comment by OYTIS** [comment]
Sorry, resubmitted the change.

**Comment by tkfu** [comment]
This doesn't work for me, either on rpi or on qemux86-64. /var/sota is not being created, and the credentials don't get copied into the image.

**Comment by OYTIS** [comment]
@tkfu Have you tried to pull [https://github.com/advancedtelematic/meta-updater/pull/72](72) as well? It was merged 10 minutes ago.

**Comment by tkfu** [comment]
@OYTIS Yes, I'd rebased off that since Wednesday. That allows the image to build, but I still don't get a /var/sota directory.

**Comment by tkfu** [comment]
@OYTIS Correction: I just ran a build off of the latest everything, and I do get a /var/sota directory, but it's empty.

**Comment by OYTIS** [comment]
@tkfu That is surprising. So you're flashing core-image-minimal-raspberrypi3.rpi-sdimg-ota, right? Does the new package appear in ATS Garage UI?

**Comment by tkfu** [comment]
@OYTIS Actually, since I don't have a Pi handy here, I'm building the qemux86-64 image. But yes, core-image-minimal.

The new package doesn't appear in ATS garage, no--it doesn't seem that the ostree_push step is being executed.

File: classes/image_types_ostree.bbclass [hunk 3]
```diff
@@ -16,6 +17,36 @@ OSTREE_KERNEL ??= "${KERNEL_IMAGETYPE}"
...
+
+        if d.getVar("OSTREE_PUSH_CREDENTIALS", True):
+            bb.warn("OSTREE_PUSH_CREDENTIALS are overriden by those in SOTA_PACKED_CREDENTIALS")
+
+        d.setVar("SOTA_AUTOPROVISION_CREDENTIALS", "%s/sota_credentials/autoprov_credentials.p12" % d.getVar("DEPLOY_DIR_IMAGE", True))
+        d.setVar("SOTA_AUTOPROVISION_URL_FILE", "%s/sota_credentials/autoprov.url" % d.getVar("DEPLOY_DIR_IMAGE", True))
+        d.setVar("OSTREE_PUSH_CREDENTIALS", "%s/sota_credentials/treehub.json" % d.getVar("DEPLOY_DIR_IMAGE", True))
+}
+
+IMAGE_DEPENDS_osreecredunpack = "unzip-native:do_populate_sysroot"
```
Comment (tkfu) [inline classes/image_types_ostree.bbclass]: Typo here, I think? Looks like it should be IMAGE_DEPENDS_ostreecredunpack

**Comment by tkfu** [comment]
PEBKAC issue: I was working off feat/zipcreds instead of feat/PRO-2945/zipcreds.

**Pull request merged**